src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.spec.ts

```diff
@@ -1,27 +1,245 @@
+        studentData: testStudent,
+        feedbackSessionLogType: LogType.FEEDBACK_SESSION_SUBMISSION,
+        timestamp: 0,
+      },
+    ],
+  };
+  const testLogs2: FeedbackSessionLog = {
+    feedbackSessionData: testFeedbackSession,
+    feedbackSessionLogEntries: [
+      {
+        studentData: testStudent,
+        feedbackSessionLogType: LogType.FEEDBACK_SESSION_SUBMISSION,
+        timestamp: 0,
+      },
+    ],
+  };
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       imports: [InstructorAuditLogsPageModule, HttpClientTestingModule],
     })
-    .compileComponents();
+        .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(InstructorAuditLogsPageComponent);
+    courseService = TestBed.inject(CourseService);
+    studentService = TestBed.inject(StudentService);
+    logService = TestBed.inject(LogService);
+    timezoneService = TestBed.inject(TimezoneService);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should snap with default fields', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when page is still loading', () => {
+    component.isLoading = true;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when searching for details in search form', () => {
+    component.courses = [testCourse1, testCourse2];
+    component.formModel = {
+      logsDateFrom: { year: 1997, month: 9, day: 11 },
+      logsTimeFrom: { hour: 23, minute: 59 },
+      logsDateTo: { year: 1998, month: 9, day: 11 },
+      logsTimeTo: { hour: 15, minute: 0 },
+      courseId: 'CS9999',
+      studentName: 'Doe John',
+    };
+    component.courseToStudents = {
+      CS9999: [testStudent],
+      MA1234: [],
+    };
+    component.isLoading = false;
+    component.isSearching = true;
+    fixture.detectChanges();
+
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap with results of a search', () => {
+    component.searchResults = [
+      {
+        feedbackSessionName: 'Feedback session 1',
+        logColumnsData: resultColumns,
+        logRowsData: [[
+          { value: '15 January 2021' },
+          { value: 'Doe John' },
+          { value: 'Viewed the submission page' },
+          { value: 'doejohn@email.com' },
+          { value: 'section 1' },
+          { value: 'team 1' },
+        ]],
+        isTabExpanded: true,
+      },
+      {
+        feedbackSessionName: 'Feedback session 2',
+        logColumnsData: resultColumns,
+        logRowsData: [],
+        isTabExpanded: true,
+      },
+    ];
+    component.isLoading = false;
+    component.isSearching = false;
+    fixture.detectChanges();
+
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should load all courses and students that instructor has on init', () => {
+    const emptyStudent: Student = {
```

Comment:
I think we can declare this outside this test case so others can reuse it in the future